source/puppet/5.3/ssl_autosign.markdown

```diff
@@ -90,27 +88,26 @@ If you aren't embedding additional data, the CSR will contain only the node's ce
 
 As you can see, you must think things through carefully when designing your CSR data and signing policy. As long as you can arrange reasonable end-to-end security for secret data on your nodes, you should be able to rig up a secure autosigning system.
 
-
 ### Policy executable API
 
 The API for policy executables is as follows:
 
-* **Run environment:** The executable will be run once for each incoming CSR.
-    * It will be executed by the Puppet master process and will run as the same user as the Puppet master.
-    * The Puppet master process will _block until the executable finishes running._ We expect policy executables to finish in a timely fashion; if they do not, it's possible for them to tie up all available Puppet master threads and deny service to other agent nodes. If an executable needs to perform network requests or other potentially expensive operations, the author is in charge of implementing any necessary timeouts, possibly bailing and exiting non-zero in the event of failure.
-    * (Note that under a Rack server like Passenger, there are generally multiple Puppet master processes available at any given time, so policy executables have a little bit of leeway.)
-* **Arguments:** The executable must allow a single command line argument. This argument will be the Subject CN (certname) of the incoming CSR.
-    * No other command line arguments should be provided.
-    * The Puppet master should never fail to provide this argument.
-* **Stdin:** The executable will receive the entirety of the incoming CSR on its stdin stream. The CSR will be encoded in PEM format.
-    * The stdin stream will contain nothing but the complete CSR.
-    * The Puppet master should never fail to provide the CSR on stdin.
-* **Exit status:** The executable must exit with a status of `0` if the certificate should be autosigned; it must exit with a non-zero status if it should not be autosigned.
-    * The Puppet master will treat all non-zero exit statuses as equivalent.
-* **Stdout and stderr:** Anything the executable emits on stdout or stderr will be copied to the Puppet master's log output at the `debug` log level. Puppet will otherwise ignore the executable's output; only the exit code is considered significant.
+-   **Run environment:** The executable will be run once for each incoming CSR.
+    -   It will be executed by the Puppet master process and will run as the same user as the Puppet master.
+    -   The Puppet master process will _block until the executable finishes running._ We expect policy executables to finish in a timely fashion; if they do not, it's possible for them to tie up all available Puppet master threads and deny service to other agent nodes. If an executable needs to perform network requests or other potentially expensive operations, the author is in charge of implementing any necessary timeouts, possibly bailing and exiting non-zero in the event of failure.
```

Comment:
I'm not sure if the nuance here matters, but a signing request will tie up a single jruby instance, so depending on how many jrubies are configured, this might not block all requests (also non-ruby requests will still be serviced).

Reply:
That could be a good note for the "differences from Puppet" page on Puppet Server.